lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -211,6 +213,7 @@ WasmBytecodeGenerator::GenerateFunction()
             {
                 throw WasmCompilationException(_u("Last expression return type mismatch return type"));
             }
+            m_reader->m_currentNode.ret.arity = 1;
             EmitReturnExpr(&exprInfo);
```

Comment:
You need to push `exprInfo` to the eval stack if the return type is not `Void` before Emitting the Return expression.